File: src/test/java/com/example/pages/DragDropPage.java
package com.example.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

public class DragDropPage extends SeleniumBasePage {

    private final By source = By.id("drag-source");
    private final By target = By.id("drop-target");

    public DragDropPage(WebDriver driver) {
        super(driver);
    }

    public DragDropPage dragAndDrop() {
        WebElement src = waitForVisible(source);
        WebElement tgt = waitForVisible(target);
        actions.dragAndDrop(src, tgt).perform();
        return this;
    }
}

File: python/pages/dragdrop_page.py
from selenium.webdriver.common.by import By
from .base_page import BasePage


class DragDropPage(BasePage):

    SOURCE = (By.ID, "drag-source")
    TARGET = (By.ID, "drop-target")

    def drag_and_drop(self):
        src = self.wait_visible(self.SOURCE)
        tgt = self.wait_visible(self.TARGET)
        self.actions.drag_and_drop(src, tgt).perform()
        return self
